samples/bluetooth/hci_ipc/overlay-nrf5340_cpunet_iso_nrf21540_ek-bt_ll_sw_split.conf

```diff
@@ -0,0 +1,6 @@
+# Reduce from 310 bytes, in nrf5340_cpunet_iso-bt_ll_sw_split.conf
+# to be able to fit in 64KB RAM, in case needed in the future.
+
+# Example:
+# CONFIG_BT_ISO_TX_MTU=247
+# CONFIG_BT_ISO_RX_MTU=251
```

Comment:
As I mentioned in https://github.com/zephyrproject-rtos/zephyr/pull/80733#discussion_r1830602527: This is messy. We shouldn't have overlays/extra conf files that need to be added in a specific order to work, especially when it's not documented.

Reply:
> OVERLAY_CONFIG is deprecated, but it did convey that something was overlay-ed!

Right :D I should stop calling them overlays

> There is repo wide use of extra_configs in sample.yaml and testcase.yaml files to use overrides of Kconfig values from prj.conf and use of EXTRA_CONF_FILE is better organization wise compared to use of list of Kconfigs under extra_configs.

If we have several places where we use multiple extra configs that needs to be specific in a specific order, that's not a good reason to do it here either. IMO that just show that we have this other places too. 

A file called `testcase.yaml` does/should not be used as documentation. If these values are required for this specific board to work, then that needs to be documented and/or applied automatically.

Reply:
It is documented  in the snippets section that the processing order of the snippets, `.conf` and `.overlay` files are "in the order they are listed".

https://github.com/zephyrproject-rtos/zephyr/blob/28796076d84056d085e5a4ebf213dafbfe56202a/doc/build/snippets/writing.rst?plain=1#L98-L126